packages/jellyfish-api-core/__tests__/category/masternode/createMasternode.test.ts

```diff
@@ -0,0 +1,64 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { MasternodeState } from '../../../src/category/masternode'
+
+describe('Masternode', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should create a masternode transaction', async () => {
+    const masternodesBefore = await client.masternode.listMasternodes()
+    const masternodesLengthBefore = Object.keys(masternodesBefore).length
+
+    const address = await client.wallet.getNewAddress()
+    const masternode = await client.masternode.createMasternode(address)
+
+    await container.generate(1)
+
+    const masternodesAfter = await client.masternode.listMasternodes()
+    const masternodesLengthAfter = Object.keys(masternodesAfter).length
+
+    expect(masternodesLengthAfter).toStrictEqual(masternodesLengthBefore + 1)
+    expect(typeof masternode).toStrictEqual('string')
+
+    for (const masternode in masternodesAfter) {
+      const createdMasternode = masternodesAfter[masternode]
+      if (createdMasternode.ownerAuthAddress === address) {
+        expect(typeof createdMasternode.ownerAuthAddress).toStrictEqual('string')
```

Comment:
I think you miss these 3 variables.

 ownerIsMine: false,
 operatorIsMine: false,
 localMasternode: false,